PWGHF/Core/HfMlResponseDstarToD0Pi.h

```diff
@@ -0,0 +1,230 @@
+/// \brief Class to compute the ML response for D*± → D0(bar) π± analysis selections
+/// \author Mingze Li <Mingze.li@cern.ch>
+
+#ifndef PWGHF_CORE_HFMLRESPONSEDSTARTOD0PI_H_
+#define PWGHF_CORE_HFMLRESPONSEDSTARTOD0PI_H_
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "PWGHF/Core/HfMlResponse.h"
+
+// Fill the map of available input features
+// the key is the feature's name (std::string)
+// the value is the corresponding value in EnumInputFeatures
+#define FILL_MAP_DSTAR(FEATURE)                                         \
+  {                                                                     \
+#FEATURE, static_cast < uint8_t>(InputFeaturesDstarToD0Pi::FEATURE) \
+  }
+
+// Check if the index of mCachedIndices (index associated to a FEATURE)
+// matches the entry in EnumInputFeatures associated to this FEATURE
+// if so, the inputFeatures vector is filled with the FEATURE's value
+// by calling the corresponding GETTER from OBJECT
+#define CHECK_AND_FILL_VEC_DSTAR_FULL(OBJECT, FEATURE, GETTER)    \
+  case static_cast<uint8_t>(InputFeaturesDstarToD0Pi::FEATURE): { \
+    inputFeatures.emplace_back(OBJECT.GETTER());                  \
+    break;                                                        \
+  }
+
+// Specific case of CHECK_AND_FILL_VEC_DSTAR_FULL(OBJECT, FEATURE, GETTER)
+// where OBJECT is named candidate and FEATURE = GETTER
+#define CHECK_AND_FILL_VEC_DSTAR(GETTER)                         \
+  case static_cast<uint8_t>(InputFeaturesDstarToD0Pi::GETTER): { \
+    inputFeatures.emplace_back(candidate.GETTER());              \
+    break;                                                       \
+  }
+
+// Very specific case of CHECK_AND_FILL_VEC_DSTAR_FULL(OBJECT, FEATURE, GETTER)
+// Use for push back different value for D*+ or D*- candidate
+#define CHECK_AND_FILL_VEC_DSTAR_CHARGEBASE(POSGETTER, NEGGETTER, FEATURENAME) \
+  case static_cast<uint8_t>(InputFeaturesDstarToD0Pi::FEATURENAME): {          \
+    if (candidate.signSoftPi() > 0) {                                          \
+      inputFeatures.emplace_back(candidate.POSGETTER());                       \
+    } else {                                                                   \
+      inputFeatures.emplace_back(candidate.NEGGETTER());                       \
+    }                                                                          \
+    break;                                                                     \
+  }
+
+namespace o2::analysis
+{
+
+enum class InputFeaturesDstarToD0Pi : uint8_t {
+  chi2PCAD0 = 0,
+  decayLengthD0,
+  decayLengthXYD0,
+  decayLengthNormalisedD0,
+  decayLengthXYNormalisedD0,
+  cpaD0,
+  cpaXYD0,
+  deltaIPNormalisedMaxD0,
+  impactParameterProductD0,
+  ptProng0,
+  ptProng1,
+  ptSoftPi,
+  impactParameter0,
+  impactParameter1,
+  impParamSoftPi,
+  impactParameterNormalised0,
+  impactParameterNormalised1,
+  normalisedImpParamSoftPi,
+  cosThetaStarD0,
+  massD0,
+  massDStar,
```

Comment:
The D* mass should never been used as a ML feature, so it would be better to remove it.